src/Renci.SshNet/Session.cs

```diff
@@ -992,6 +992,7 @@ internal void WaitOnHandle(WaitHandle waitHandle, TimeSpan timeout)
             switch (signaledElement)
             {
                 case 0:
+                    System.Runtime.ExceptionServices.ExceptionDispatchInfo.Capture(_exception).Throw();
                     throw _exception;
```

Comment:
```suggestion
                    break;
```

Reply:
Feel free to ignore that 🙂

Reply:
I was thinking the same thing, even though it's unused code anyway.